smurfFinalProject/smurfFinalProject/node.cpp

```diff
@@ -0,0 +1,168 @@
+    return variable;
+}
+
+int variableNode::accept(visitor *visitorx){                            //Accepts the interpreter for a variable
+    return visitorx->evaluate_variable(this, variable);
+    
+}
+
+void variableNode::print(){                                             //Prints the variable
+    cout<<"Prining Variable Node | Variable: "<<variable<<endl;
+}
+
+//assignment node––––––––––––––––––––––––––––––––––––––––
+assignmentNode::assignmentNode(node *l, node *r){                       //Node tree structure to initialize a variable. As mentioned
+    left = l;                                                           //in main, this structure consists of one parent assignment
+    assignment = "=";                                                   //node and two chidren nodes containing a variable and value
+    right = r;
+    string lstr = l->str();
+    string rstr = r->str();
+    //print(lstr, rstr);
+}
+
+string assignmentNode::str(){                                           //Returns the assignment in a string
+    return "(" + left->str()+ "=" + right->str() + ")";
+}
+
+int assignmentNode::accept(visitor *visitorx){                          //Accepts the interpreter for an assignment
+    return visitorx->evaluate_assignment(this, left, right);
+}
+
+void assignmentNode::print(string l, string r) {                        //Prints the assignment
+    cout<<"Printing Assignment Node | Assignment: = | Left: "<<l<<" | Right: "<<r<<endl;
+}
+
+//statement node–––––––––––––––––––––––––––––––––––––––––
+statementNode::statementNode(node* stmt){                               //Node to hold statements (could consist of any previous nodes)
+    statement = stmt;
+    print();
+}
+
+string statementNode::str(){                                            //Returns the node in a string
+    return statement->str();
+}
+
+int statementNode::accept(visitor *visitorx){                           //Accepts the interpreter for a statement
+    cout<<"Statement: "<<statement[0].str()<<endl;
+    return visitorx->evaluate_statement(statement);
+}
+
+void statementNode::print(){                                            //Prints the statement
+    cout<<"Printing Statement Node | Statement: "<<statement->str()<<endl;
+}
+
+//code node––––––––––––––––––––––––––––––––––––––––––––––
+codeNode::codeNode(){                                                   //Parent node to the whole AST
+    print();
+}
+
+int codeNode::accept(visitor *visitorx){                                //Accepts the interpreter for code
+    return visitorx->evaluate_code(statements);
+}
+
+string codeNode::str(){                                                 //Returns the statements in strings
+    string result;
+    for(int i=0; i<statements.size(); i++){
+        result = statements[i]->str();
+    }
+    return result;
+}
+
+void codeNode::addToVect(node *nodex){                                  //Adds statements to the statements vector of nodes
```

Comment:
Most of your naming is great. But this one is suspect: `addToVect` describes the implementation, not the intent. Maybe `addStatement`?